integrations/tensorflow/e2e/math/math_test.py

```diff
@@ -0,0 +1,561 @@
+        configure_dtypes(Config(signature=BINARY), ALL_NUMBER_TYPES),
+    "subtract":
+        configure_dtypes(Config(signature=BINARY), ALL_NUMBER_TYPES),
+    "tan":
+        configure_dtypes(Config(signature=UNARY), FLOATING_NUMBER_TYPES),
+    "tanh":
+        configure_dtypes(Config(signature=UNARY), FLOATING_NUMBER_TYPES),
+    "top_k":
+        Config(signature=UNARY_FLOAT, kwargs=dict(k=2)),
+    "truediv":
+        configure_dtypes(Config(signature=BINARY), FLOATING_NUMBER_TYPES),
+    "unsorted_segment_max":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_mean":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_min":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_prod":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_sqrt_n":
+        UNSORTED_SEGMENT_CONFIG,
+    "unsorted_segment_sum":
+        UNSORTED_SEGMENT_CONFIG,
+    "xdivy":
+        configure_dtypes(Config(signature=BINARY), FLOATING_NUMBER_TYPES),
+    "xlog1py":
+        configure_dtypes(Config(signature=BINARY), FLOATING_NUMBER_TYPES),
+    "xlogy":
+        configure_dtypes(Config(signature=BINARY), FLOATING_NUMBER_TYPES),
+    "zero_fraction":
+        configure_dtypes(Config(signature=UNARY), ALL_NUMBER_TYPES),
+    "zeta":
+        Config(
+            signature=BINARY_FLOAT,
+            # The function is poorly behaved near zero, so we test this range
+            # to avoid outputing all nans.
+            generator=lambda *args: tf_utils.uniform(*args, low=3.0, high=4.0)),
+}
+
+# Normalize FUNCTION_TO_CONFIGS as described above.
+for function_name, configs in FUNCTION_TO_CONFIGS.items():
+  if isinstance(configs, Config):
+    FUNCTION_TO_CONFIGS[function_name] = {function_name: configs}
+  else:
+    # Prepend the function names to existing exported names to make the logs
+    # readable.
+    FUNCTION_TO_CONFIGS[function_name] = {
+        f"{function_name}_{config_name}": config
+        for config_name, config in FUNCTION_TO_CONFIGS[function_name].items()
+    }
+
+flags.DEFINE_list(
+    "functions", "abs",
+    f"Any of {list(FUNCTION_TO_CONFIGS.keys())}. If more than one function is "
+    "provided then len(--target_backends) must be one.")
+flags.DEFINE_bool(
+    "dynamic_dims", False,
+    "Whether or not to compile the layer with dynamic dimensions.")
+flags.DEFINE_bool(
+    "test_complex", False,
+    "Whether or not to test or ignore function signatures with complex types.")
+flags.DEFINE_bool(
+    'list_functions_with_complex_tests', False,
+    'Whether or not to print out all functions with complex inputs '
+    '(and skip running the tests).')
+
+
+def _complex_wrapper(function):
+  """Wraps a tf.function to allow compiling functions of complex numbers."""
+
+  def decorator(*args, **kwargs):
+    inputs = []
+    for real, imag in zip(args[::2], args[1::2]):
+      inputs.append(tf.complex(real, imag))
+    result = function(*inputs, **kwargs)
+    # Directly returning a complex number causes an error.
```

Comment:
TODO? Issue?

Reply:
I'm unsure if this is done for a good reason or not. I mostly just followed what the few complex-number tests we had did. It makes sense from the perspective of serializable inputs and outputs through.